js/eqonex.js

```diff
@@ -193,36 +193,67 @@ module.exports = class eqonex extends Exchange {
         //     }
         //
         const id = this.safeString (market, 'instrumentId');
-        const uppercaseId = this.safeString (market, 'symbol');
-        const assetType = this.safeString (market, 'assetType');
-        const spot = (assetType === 'PAIR');
-        const swap = (assetType === 'PERPETUAL_SWAP');
-        const type = swap ? 'swap' : 'spot';
         const baseId = this.safeString (market, 'currency');
         const quoteId = this.safeString (market, 'contAmtCurr');
+        const settleId = this.safeString (market, 'settlCurrency');
         const base = this.safeCurrencyCode (baseId);
         const quote = this.safeCurrencyCode (quoteId);
-        const symbol = swap ? uppercaseId : (base + '/' + quote);
+        const settle = this.safeCurrencyCode (settleId);
+        const assetType = this.safeString (market, 'assetType');
+        const spot = (assetType === 'PAIR');
+        const swap = (assetType === 'PERPETUAL_SWAP');
+        let symbol = base + '/' + quote;
+        const uppercaseId = this.safeString (market, 'symbol');
+        let type = 'spot';
+        let linear = undefined;
+        let inverse = undefined;
+        let contract = false;
+        if (swap) {
+            symbol = symbol + ':' + settle;
+            type = 'swap';
+            linear = quote === settle;
+            inverse = !linear;
+            contract = true;
+        } else if (!spot) {
+            symbol = uppercaseId;
+            type = assetType;
+            contract = true;
+        }
```

Comment:
I know that right now there's only spot and swap, but I thought this was safer in case they ever add more types